partner_base_multicompany_default/models/partner.py

```diff
@@ -0,0 +1,147 @@
+# Copyright 2023 Moduon Team S.L.
+# Copyright 2024 Camptocamp SA
+# License LGPL-3.0 or later (https://www.gnu.org/licenses/lgpl-3.0)
+import logging
+from collections import defaultdict
+from functools import reduce
+from itertools import groupby
+from operator import itemgetter, or_
+
+from odoo import _, models
+from odoo.exceptions import UserError
+from odoo.osv import expression
+
+_logger = logging.getLogger(__name__)
+
+
+class Partner(models.Model):
+    """Adds helpers to create default values across companies
+
+    For each M2O property field for records that are not shared, define:
+
+    def _get_multicompany_account_account_key(self):
+        return "code"
+
+    Where account_account is the model and code is the key on which you want
+    to match a similar record on an other company.
+
+    Also add a method like
+
+    def propagate_multicompany_account_payable(self):
+        self._propagate_multicompany_field("property_account_payable_id")
+
+    That you can add in the view to let the user trigger the synchronisation for
+    a specific field.
+
+    """
+
+    _inherit = "res.partner"
+
+    def _get_multicompany_domain_by_key(self, field, key):
+        # eg. _get_multicompany_account_account_key
+        return [
+            (
+                key,
+                "in",
+                self.mapped(f"{field}.{key}"),
+            ),
+        ]
+
+    def _propagate_multicompany_m2o_by_key(self, field):
+        """Set the corresponding record accross all companies
+        based on a domain to match record on other comanies
+
+        eg. account are matched by code
+        """
+        companies = self._get_alien_companies()
+        comodel = self._fields[field].comodel_name
```

Comment:
so `field` is a field_name actually and not a field instance

Reply:
I'll precise this by adding the typing of the field and proper function description. In that context there is no reason to work with a field object. The field name is still what defines a field, thus the naming of the variable `field` seems legit to me.